### ISSUE TITLE
Test integrate SVG icons

### ISSUE THREAD

**Issue by mbastian** [gephi/gephi#2585]
Since Netbeans Platform 11, SVG icons are supported. This seems like the most scalable solution to support hi-res icons. Batik is used behind the scene, which is the same library we use for SVG export. 

### Definition of done
- Have a few high-priority icons designed and exported as SVG files
- Replace existing icons to SVG icons using the new API (see resources)
- Make sure those icons adapt well in Dark Mode (not sure exactly how this should be done yet - via other icons or automatically?)
- Documentation is written to guide future migration of all Gephi icons

Reminder: We should make sure we stay in sync with Netbeans Platform on the Batik version to avoid issues.

### Resources
- https://issues.apache.org/jira/browse/NETBEANS-2604
- https://netbeans.apache.org/wiki/DevFaqImagesForDarkLaf.asciidoc

**Comment by mbastian** [comment]
Update:
- I tried to load an SVG icon and it works. For instance
```
Icon icon = ImageUtilities.loadImageIcon("/org/gephi/visualization/component/rectangle.svg", false);
```
- I tried to provide two icons `rectange.svg` and `rectangle_dark.svg` and Netbeans switch automatically when changing the LaF from Light to Dark.
- Netbeans has a `DarkIconFilter` class that seems appropriate to automatically switch but I couldn't find it enabled in FlatLaf so probably not an option at the moment.

**Comment by mbastian** [comment]
I was able to update a few icons in the Toolbar, see

![Image](https://user-images.githubusercontent.com/234813/188122320-afcc2ee3-c675-4e04-97c4-e16cd38e9d46.png)

Notes:
- Those icons are unique and could support both the light and dark mode. It's. not ideal though in light mode and we should ideally have both to be well visible.
- I migrated all calls to icons (including PNG) to the new API `ImageUtilities.loadImageIcon`. Seems to work fine. This will make the transition to SVG easier.
- I had some issues with icon sizes, most likely due to rounding issues. NB was showing some icons 32x32 instead of 16x16. I manually changed the SVG width/height from 16 to 15.99 and it worked. We should do further tests here.
- It will be cumbersome to update all icons as they are scattered in each modules and sub-packages. We should evaluate whether it's feasible and desirable to have a unique Icons module.

Closing this one as the investigation is done. I'll open a follow-up one thought.
